### PR TITLE
Getter for trajectories in AssetsWriter

### PR DESCRIPTION
**Pull request by schwoere**
Added a getter for the trajectories of the pose graph to the AssetsWriter and moved CreateFileWriterFactory outside of the AssetsWriter class.

To register new points processors, PointsProcessors need to be constructed from outside the AssetsWriter. This may require the trajectories of the pose graph. Therefore, the getter for the trajectories was added to the AssetsWriter.

The static class method CreateFileWriterFactory was moved outside the AssetsWriter to communicate its independence of class internals of the AssetsWriter.

### CODE REVIEW

File: cartographer_ros/cartographer_ros/assets_writer.h [hunk 1]
```diff
@@ -58,6 +60,10 @@ class AssetsWriter {
       point_pipeline_builder_;
 };
 
+// Creates a FileWriterFactory which creates a FileWriter for storing assets.
+::cartographer::io::FileWriterFactory CreateFileWriterFactory(
```
Comment (wohe) [inline cartographer_ros/cartographer_ros/assets_writer.h]: Why was that changed? Seemed perfectly fine to me.

Comment (schwoere) [reply cartographer_ros/cartographer_ros/assets_writer.h]: Okay, I thought it helps to communicate the independence from AssetsWriter class internals. I reverted back.

**Comment by cschuet** [comment]
@schwoere Can we close this for now? I don't see any activity on this.

**Comment by schwoere** [comment]
@cschuet Can be closed, yes. Thanks!
